quinn-proto/src/transport_parameters.rs

```diff
@@ -371,22 +394,29 @@ mod test {
     #[test]
     fn coding() {
         let mut buf = Vec::new();
-        let params = TransportParameters {
-            initial_max_bidi_streams: 16,
-            initial_max_uni_streams: 16,
-            ack_delay_exponent: 2,
+        
+        let config = Config {
+            max_remote_bi_streams: 16,
+            max_remote_uni_streams: 16,
             max_packet_size: 1200,
+            ..Config::default()
+        };
+
+        let server_params = ServerTransportParameters {
+            params: TransportParameters::new(&config),
             preferred_address: Some(PreferredAddress {
                 address: SocketAddr::new(IpAddr::V4([127, 0, 0, 1].into()), 42),
                 connection_id: ConnectionId::new(&[]),
                 stateless_reset_token: [0xab; 16],
             }),
-            ..TransportParameters::default()
+            original_connection_id: None,
+            stateless_reset_token: None,
         };
-        params.write(Side::Server, &mut buf);
+
+        server_params.write(&mut buf);
         assert_eq!(
-            TransportParameters::read(Side::Client, &mut buf.into_buf()).unwrap(),
-            params
+            ClientTransportParameters::read(&mut buf.into_buf()).unwrap().params,
```

Comment:
`ServerTransportParameters::read` is the function that would decode an encoded `ServerTransportParameters` value.